src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -121,24 +123,22 @@ export class CellExecution {
         private readonly errorHandler: IDataScienceErrorHandler,
         private readonly editorProvider: INotebookEditorProvider,
         private readonly applicationService: IApplicationShell,
-        private readonly isPythonKernelConnection: boolean
+        private readonly isPythonKernelConnection: boolean,
+        extensionContext: IExtensionContext
     ) {
         // These are only used in the tests.
         // See where this is used to understand its purpose.
-        let deferred = createDeferred<void>();
         if (
             !CellExecution.cellsCompletedForTesting.has(cell) ||
             CellExecution.cellsCompletedForTesting.get(cell)!.completed
         ) {
-            CellExecution.cellsCompletedForTesting.set(cell, deferred);
-        }
-        if (
-            CellExecution.cellsCompletedForTesting.has(cell) &&
-            !CellExecution.cellsCompletedForTesting.get(cell)!.completed
-        ) {
-            deferred = CellExecution.cellsCompletedForTesting.get(cell)!;
+            CellExecution.cellsCompletedForTesting.set(cell, createDeferred<void>());
+        } else {
+            traceErrorIf(
```

Comment:
If this is hit, I'd like to see the logs (error so that it stands out & only in dev or testing)